.github/workflows/CI.yml

```diff
@@ -23,25 +23,35 @@ jobs:
       matrix:
         julia-version:
           - '1.6'
-          - '1.9'
-          - '~1.10.0-0'
+          - '1.9'   # to be removed in the near future
+          - '1.10'
           - 'nightly'
         group: [ 'short', 'long' ]
         os:
           - ubuntu-latest
         depwarn: [ '' ]
         include:
           # Add a single job per group with deprecation errors on the most recent stable julia version
-          - julia-version: '1.9'
-            os: ubuntu-latest
+          - julia-version: '1.10'
             group: 'short'
-            depwarn: 'depwarn=error'
-          - julia-version: '1.9'
             os: ubuntu-latest
+            depwarn: 'depwarn=error'
+          - julia-version: '1.10'
             group: 'long'
+            os: ubuntu-latest
             depwarn: 'depwarn=error'
           # Add a few macOS jobs (not too many, the number we can run in parallel is limited)
+          - julia-version: '1.10'
+            group: 'short'
+            os: macOS-latest
+          - julia-version: '1.10'
+            group: 'long'
+            os: macOS-latest
           - julia-version: 'nightly'
+            group: 'short'
+            os: macOS-latest
+          - julia-version: 'nightly'
+            group: 'long'
```

Comment:
Do I understand it correctly that this makes us go from 1 to 4 macOS jobs? (Maybe not, I don't know what the "group: short/long" does, so maybe only 2 out of 4 are run?)

Point is, last I checked we can only run 5 macOS jobs concurrently over all of `oscar-system`. So instead of being able to complete about 4-5 full PR builds in parallel, with this patch we could only do 1.25 (?), meaning that completion of our macOS CI jobs would become again a major bottleneck.

Perhaps it suffices to just test with 1.10, not against nightly? Or do one test against 1.10 and one against nightly?


Perhaps @aaruni96 who is looking into setting up a GitHub CI runner in Kaiserslautern can also look into how feasible it is to add a Mac as CI runner (we could easily buy a Mac mini or Mac studio)

Reply:
We are testing against stable releases to keep the correctness of our code and notice regressions on our side.
For nightly, this is different: here we test mainly against to notice regression on the julia side. In my opinion, the latter doesn't need to be tested on every commit of every PR. It would be enough to e.g. have an automatic daily (?) run of our master against nightly to notice regressions. Most regressions and breakages are recorded by the ubuntu runners anyway.

Reply:
> [...] It would be enough to e.g. have an automatic daily (?) run of our master against nightly to notice regressions. Most regressions and breakages are recorded by the ubuntu runners anyway.

We have tried this and it does not work in practice.

Reply:
"daily runs" only work if a bunch of people watches those closely -- but that's difficult to ensure. One way could be to set up Slack integration (as we've done for the GAP.jl CI jobs) so that failed jobs ping there -- but that still is far inferior to having this directly in the PR jobs...

That said, it may be sufficient to have one nightly job, as usually the Julia nightly regressions manifest as "Oscar does not start anymore" / "Oscar quickly crashes" ?

Anyway, none of this is a strong objection to this PR -- if you guys thought this through, let's do it, be it with 3 or with 4 macOS jobs. We'll just tweak it as we go along. And we should look into getting our own macOS runners, too...